integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/output/DataQualityWriter.java

```diff
@@ -0,0 +1,192 @@
+    public List<DataQualityRow> getDataQualityRows() {
+        return dqRows;
+    }
+
+
+    /**
+     * Get count of output rows
+     *
+     * @return row count
+     */
+    public int getDataQualityRowCount() {
+        return dqRows.size();
+    }
+
+    /**
+     * Set Spark context
+     *
+     * @param sc JavaSparkContext
+     */
+    public void setSparkContext(JavaSparkContext sc) {
+        this.sc = sc;
+    }
+
+    /**
+     * Set Hive context
+     *
+     * @param hiveContext HiveContext
+     */
+    public void setHiveContext(HiveContext hiveContext) {
+        this.hiveContext = hiveContext;
+    }
+
+    /**
+     * Add row to write in output
+     *
+     * @param row row for output
+     */
+    public void addRow(DataQualityRow row) {
+        dqRows.add(row);
+    }
+
+    /**
+     * Add multiple rows to write in output
+     *
+     * @param rows list of rows for output
+     */
+    public void addRows(List<DataQualityRow> rows) {
+        dqRows.addAll(rows);
+    }
+
+    /**
+     * Write result to Hive table
+     *
+     * @param scs SparkContextService
+     * @param outputTable Name of the destination Hive table
+     * @param partitionValue Value of the partition for Hive. Currently the processing_dttm column
+     * @return boolean indicating result of write
+     */
+    public boolean writeResultToTable(SparkContextService scs, String outputTable, String partitionValue) {
+
+        boolean retVal = false;
+
+        if (sc == null) {
+            log.error("Error writing result: Spark context is not available.");
+        } else if (hiveContext == null) {
+            log.error("Error writing result: Hive context is not available.");
+        } else if (dqRows.isEmpty()) {
+            log.info("Error writing results: No data to write");
+        } else {
+
+            JavaRDD<DataQualityRow> dqRowsRDD = sc.parallelize(dqRows);
+            DataSet dqRowsDF = scs.toDataSet(hiveContext.createDataFrame(dqRowsRDD, DataQualityRow.class));
+
+            String tempTable = "DQ_DF_TABLE";
+            dqRowsDF.registerTempTable(tempTable);
+
+            createOutputTableIfNotExists(scs, outputTable);
+            writeResultToOutputTable(scs, tempTable, outputTable, partitionValue);
+
+            retVal = true;
+        }
+
+        return retVal;
+    }
+
+    /**
+     * Creates target Hive table if it does not already exist
+     * 
+     * @param scs SparkContextService
+     * @param outputTable Name of the destination Hive table
+     */
+    private void createOutputTableIfNotExists(SparkContextService scs, String outputTable) {
+        String createTableSQL = "CREATE TABLE IF NOT EXISTS " + outputTable
+                                + "\n"
+                                + "(rule_name STRING, rule_description STRING, status BOOLEAN, rule_resultdetail STRING)\n"
+                                + "PARTITIONED BY ( processing_dttm STRING)\n";
+
+        log.info("Executing: " + createTableSQL);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2629)